Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/FilterAdapter.cpp

```diff
@@ -0,0 +1,501 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include <AzFramework/DocumentPropertyEditor/AdapterBuilder.h>
+#include <AzToolsFramework/UI/DocumentPropertyEditor/FilterAdapter.h>
+
+namespace AZ::DocumentPropertyEditor
+{
+    RowFilterAdapter::RowFilterAdapter() 
+    {
+    }
+
+    void RowFilterAdapter::SetSourceAdapter(DocumentAdapterPtr sourceAdapter)
+    {
+        m_sourceAdapter = sourceAdapter;
+        m_resetHandler = DocumentAdapter::ResetEvent::Handler(
+            [this]()
+            {
+                this->HandleReset();
+            });
+        m_sourceAdapter->ConnectResetHandler(m_resetHandler);
+
+        m_changedHandler = DocumentAdapter::ChangedEvent::Handler(
+            [this](const Dom::Patch& patch)
+            {
+                this->HandleDomChange(patch);
+            });
+        m_sourceAdapter->ConnectChangedHandler(m_changedHandler);
+
+        m_domMessageHandler = AZ::DocumentPropertyEditor::DocumentAdapter::MessageEvent::Handler(
+            [this](const AZ::DocumentPropertyEditor::AdapterMessage& message, AZ::Dom::Value& value)
+            {
+                this->HandleDomMessage(message, value);
+            });
+        m_sourceAdapter->ConnectMessageHandler(m_domMessageHandler);
+
+        // populate the filter data from the source's full adapter contents, just like a reset
+        HandleReset();
+    }
+
+    bool RowFilterAdapter::IsRow(const Dom::Value& domValue)
+    {
+        return (domValue.IsNode() && domValue.GetNodeName() == Dpe::GetNodeName<Dpe::Nodes::Row>());
+
+    }
+
+    bool RowFilterAdapter::IsRow(const Dom::Path& sourcePath) const
+    {
+        auto sourceRoot = m_sourceAdapter->GetContents();
```

Comment:
nit: DOM copies are quite cheap compared to the lookup but it's valid to `const auto&` these guys and avoid it entirely here

Reply:
thanks! Fixed in multiple places now.